nflog/nflog.go

```diff
@@ -289,48 +216,79 @@ func marshalMeshEntry(e *pb.MeshEntry) ([]byte, error) {
 	return buf.Bytes(), nil
 }
 
+// Options configures a new Log implementation.
+type Options struct {
+	SnapshotReader io.Reader
+	SnapshotFile   string
```

Comment:
Let's add a doc-comment to each, saying only one of these fields should be set. That way, callers can see this in their editors directly - just makes it a little more convenient to use.

Reply:
The top-level comment reads as:

```
// A snapshot file or reader from which the initial state is loaded.
// None or only one of them must be set.
```

In my editor it shows as:

<img width="517" alt="Screenshot 2023-01-19 at 15 23 58" src="https://user-images.githubusercontent.com/231583/213482334-a8f6fae7-6930-4b89-a471-567beb29140d.png">


Which indicates:

>// None or only one of them must be set.

Are you thinking of something different? Happy to do it, but unsure of what this means for a different editor 🤔